dev_tools/utils/nxdl_utils.py

```diff
@@ -67,43 +85,34 @@ def get_hdf_parent(hdf_info):
 
 def get_parent_path(hdf_name):
     """Get parent path"""
-    return "/".join(hdf_name.split("/")[:-1])
+    return hdf_name.rsplit("/", 1)[0]
 
 
 def get_hdf_info_parent(hdf_info):
     """Get the hdf_info for the parent of an hdf_node in an hdf_info"""
     if "hdf_path" not in hdf_info:
         return {"hdf_node": hdf_info["hdf_node"].parent}
-    node = (
-        get_hdf_root(hdf_info["hdf_node"])
-        if "hdf_root" not in hdf_info
-        else hdf_info["hdf_root"]
-    )
-    for child_name in hdf_info["hdf_path"].split("/")[1:-1]:
-        node = node[child_name]
+    node = get_hdf_parent(hdf_info)
     return {"hdf_node": node, "hdf_path": get_parent_path(hdf_info["hdf_path"])}
 
 
 def get_nx_class(nxdl_elem):
     """Get the nexus class for a NXDL node"""
-    if "category" in nxdl_elem.attrib.keys():
-        return None
-    try:
-        return nxdl_elem.attrib["type"]
-    except KeyError:
-        return "NX_CHAR"
+    if "category" in nxdl_elem.attrib:
+        return ""
```

Comment:
Indeed, this can be an issue, and is an issue, because others were checking for being None. Now they had to be changed to check for being empty string, too.